compiler/src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -824,16 +813,20 @@ object TreeChecker {
         && !isPrimaryConstructorReturn
         && !pt.isInstanceOf[FunOrPolyProto]
       then
-        assert(tree.tpe <:< pt, {
```

Comment:
The proposed changes will also add the following check here:  `tree.tpe eq pt`. Is that ok to add ?

Reply:
Yep